pyomo/contrib/appsi/solvers/tests/test_highs_persistent.py

```diff
@@ -80,6 +80,43 @@ def test_mutable_params_with_remove_vars(self):
         res = opt.solve(m)
         self.assertAlmostEqual(res.best_feasible_objective, -9)
 
+    def test_fix_and_unfix(self):
+        # Tests issue https://github.com/Pyomo/pyomo/issues/3127
+
+        m = pe.ConcreteModel()
+        m.x = pe.Var(domain = pe.Binary)
+        m.y = pe.Var(domain = pe.Binary)
+        m.fx = pe.Var(domain = pe.NonNegativeReals)
+        m.fy = pe.Var(domain = pe.NonNegativeReals)
+        m.c1 = pe.Constraint(expr = m.fx <= m.x)
+        m.c2 = pe.Constraint(expr = m.fy <= m.y)
+        m.c3 = pe.Constraint(expr = m.x + m.y <= 1)
+
+        m.obj = pe.Objective(
+            expr = m.fx * 0.5 + m.fy * 0.4, 
+            sense = pe.maximize,
+        )
+
+        opt = Highs()
+
+        # solution 1 has m.x == 1 and m.y == 0
```

Comment:
It would be good to test the actual result returned by HiGHS here and after the next two solves.  Something like
```python
self.assertAlmostEqual(m.fx.value, 1, places=5)
self.assertAlmostEqual(m.fy.value, 0, places=5)
```